### DISCUSSION TITLE
Renovate suggests upgrades that aren't compatible with maven.compiler.source/target

### DISCUSSION THREAD

**Discussion by dblock** [renovatebot/renovate#16448] [Suggest an Idea]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

In https://github.com/acm19/aws-request-signing-apache-interceptor/pull/26 renovate recommended upgrading [com.puppycrawl.tools:checkstyle](https://checkstyle.org/) from 9.3 -> 10.3.1, however version 10 is not compatible with JDK 8 and breaks CI. Feels like renovate should be aware of this and either not suggest the upgrade, or suggest an upgrade for JDK first.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by viceice** [comment]
it's not a bug. renovate doesn't support that.

**Reply by dblock** [reply]
Thanks! It's a feature request then.

What do you suggest I do for https://github.com/acm19/aws-request-signing-apache-interceptor/pull/26? How do I prevent renovate from upgrading the library in question so that it stops making failing PRs?

**Reply by viceice** [reply]
use dashboard approval for major version updates

**Reply by rarkins** [reply]
Either upgrade JDK or set a rule to ignore those dependencies.
